lwgenerate/gcp/gcp.go

```diff
@@ -448,26 +482,116 @@ func createLaceworkProvider(laceworkProfile string) (*hclwrite.Block, error) {
 	return nil, nil
 }
 
-func createGcpProvider(serviceAccountCredentials string, projectId string) ([]*hclwrite.Block, error) {
-	attrs := map[string]interface{}{}
+func createGcpProvider(
+	serviceAccountCredentials string,
+	projectId string,
+	regionsArg []string,
+) ([]*hclwrite.Block, error) {
+	blocks := []*hclwrite.Block{}
 
-	if serviceAccountCredentials != "" {
-		attrs["credentials"] = serviceAccountCredentials
+	regions := append([]string{}, regionsArg...)
+	if len(regions) == 0 {
+		regions = append(regions, "")
 	}
 
-	if projectId != "" {
-		attrs["project"] = projectId
+	for _, region := range regions {
+		attrs := map[string]interface{}{}
+
+		if serviceAccountCredentials != "" {
+			attrs["credentials"] = serviceAccountCredentials
+		}
+
+		if projectId != "" {
+			attrs["project"] = projectId
+		}
+
+		if region != "" {
+			attrs["alias"] = region
+			attrs["region"] = region
+		}
+
+		provider, err := lwgenerate.NewProvider(
+			"google",
+			lwgenerate.HclProviderWithAttributes(attrs),
+		).ToBlock()
+		if err != nil {
+			return nil, err
+		}
+
+		blocks = append(blocks, provider)
 	}
 
-	provider, err := lwgenerate.NewProvider(
-		"google",
-		lwgenerate.HclProviderWithAttributes(attrs),
-	).ToBlock()
-	if err != nil {
-		return nil, err
+	return blocks, nil
+}
+
+func createAgentless(args *GenerateGcpTfConfigurationArgs) ([]*hclwrite.Block, error) {
+	if !args.Agentless {
+		return nil, nil
+
+	}
+
+	blocks := []*hclwrite.Block{}
+
+	regions := append([]string{}, args.Regions...)
+	if len(regions) == 0 {
+		regions = append(regions, "")
 	}
 
-	blocks := []*hclwrite.Block{provider}
+	for i, region := range regions {
+		moduleName := "lacework_gcp_agentless_scanning_global"
+		moduleDetails := []lwgenerate.HclModuleModifier{
+			lwgenerate.HclModuleWithVersion(lwgenerate.GcpAgentlessVersion),
+		}
+
+		attributes := map[string]interface{}{"regional": true}
+		if i == 0 {
+			attributes["global"] = true
+			if len(args.ProjectFilterList) > 0 {
+				projectIDs := []string{}
+				for _, id := range args.ProjectFilterList {
+					projectIDs = append(projectIDs, fmt.Sprintf("\"%s\"", id))
+				}
+				attributes["project_filter_list"] = lwgenerate.CreateSimpleTraversal([]string{
+					fmt.Sprintf("[%s]", strings.Join(projectIDs, ", ")),
+				})
+			}
+			if args.OrganizationIntegration {
+				attributes["integration_type"] = "ORGANIZATION"
+				attributes["organization_id"] = args.GcpOrganizationId
+			}
+		}
+		if i > 0 {
+			moduleName = "lacework_gcp_agentless_scanning_region_" + region
+			attributes["global_module_reference"] = lwgenerate.CreateSimpleTraversal(
+				[]string{"module", "lacework_gcp_agentless_scanning_global"},
+			)
+		}
+
+		if region != "" {
```

Comment:
Similar question to [L537](https://github.com/lacework/go-sdk/pull/1500/files#diff-9406e9de58811734965d500cfe472be2bc13f42a986ddf34ce16f8817ec298feR537); why would we want an empty region value? Should that be an error?